### PR TITLE
Floating point symfpu support 3

### PR DESCRIPTION
**Pull request by martin-cs**
(includes the previous two support commits, only depends on the first)

Adds 5 new kinds so that later code can convert partially defined functions into totally defined ones.

### CODE REVIEW

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 1]
```diff
@@ -410,6 +411,67 @@ namespace constantFold {
...
+
+    } else {
+      FloatingPoint::PartialFloatingPoint res(arg1.min(arg2));
+
+      if (res.second) {
+	Node lit = NodeManager::currentNM()->mkConst(res.first);
+	return RewriteResponse(REWRITE_DONE, lit);
+      } else {
+	// Can't constant fold the underspecified case
+	return RewriteResponse(REWRITE_DONE, node);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: Just out of curiosity. Why not do something similar to what you do with the rounding modes where you list all possible outcomes? 

https://github.com/martin-cs/CVC4/blob/floating-point-symfpu/src/theory/fp/theory_fp_rewriter.cpp#L1269-L1282

**Comment by martin-cs** [comment]
On Wed, 2017-09-27 at 18:36 +0000, Andres Noetzli wrote:
> 4tXJ7f approved this pull request.
> 
> LGTM
> 
> > +    if (node[2].getMetaKind() == kind::metakind::CONSTANT) {
> +      BitVector arg3(node[2].getConst<BitVector>());
> +
> +      FloatingPoint folded(arg1.minTotal(arg2, arg3.isBitSet(0)));
> +      Node lit = NodeManager::currentNM()->mkConst(folded);
> +      return RewriteResponse(REWRITE_DONE, lit);
> +
> +    } else {
> +      FloatingPoint::PartialFloatingPoint res(arg1.min(arg2));
> +
> +      if (res.second) {
> +	Node lit = NodeManager::currentNM()->mkConst(res.first);
> +	return RewriteResponse(REWRITE_DONE, lit);
> +      } else {
> +	// Can't constant fold the underspecified case
> +	return RewriteResponse(REWRITE_DONE, node);
> 
> Just out of curiosity. Why not do something similar to what you do with the rounding modes where you list all possible outcomes? 
> 
> https://github.com/martin-cs/CVC4/blob/floating-point-symfpu/src/theory/fp/theory_fp_rewriter.cpp#L1269-L1282

Good question ... probably because I wrote this before the
constant(-ish) folding with rounding-modes and I was thinking of doing
this in a consistent way with the other (formerly) partial functions.

**Comment by 4tXJ7f** [comment]
Makes sense. I opened issue #1160 to remember this.

**Comment by martin-cs** [comment]
Rebased to get this up to date.

**Pull request merged**